service/storage_service.hh

```diff
@@ -190,8 +190,8 @@ private:
     // Update pending ranges locally and then replicate to all cores.
     // Should be serialized under token_metadata_lock.
     // Must be called on shard 0.
-    future<> update_pending_ranges(mutable_token_metadata_ptr tmptr, sstring reason);
-    future<> update_pending_ranges(sstring reason, acquire_merge_lock aml = acquire_merge_lock::yes);
+    future<> update_pending_ranges(mutable_token_metadata_ptr tmptr, sstring reason, bool request_read_new = false);
+    future<> update_pending_ranges(sstring reason, acquire_merge_lock aml = acquire_merge_lock::yes, bool request_read_new = false);
```

Comment:
This default seems unsafe, because a later update_pending_ranges() which misses the current value can wipe out request_read_new.

Reply:
ok

Reply:
Ah, I confused it with `token_metadata::update_pending_ranges`, I have no default for it there.

I don't want to remove default here:
* there are a lot of usages for this `update_pending_ranges`, in all of them this parameter is irrelevant and it only will make a noise if we state false explicitly. It only matters when transitioning to `write_both_read_new`, it's 3 usages out of total of two dozens over `storage_service`.
* we will have to move `request_read_new` before `acquire_merge_lock` for the second overload, this will make unnecessary changes

Reply:
> It only matters when transitioning to write_both_read_new

But once we transition to write_both_read_new, shouldn't request_read_new=true stick to the token metadata until revoked, even in later transitions? There will be more transitions after the transition to write_both_read_new. 

Maybe it should be a flag in token metadata which update_pending_ranges() picks up. This way we also avoid modifying places which don't care about it.

Reply:
Now I see your point, and it's the argument I missed when @gleb-cloudius convinced me to do the opposite, initially there was a separate method to set this flag.

@gleb-cloudius are those numerous gossiper-induced `update_pending_ranges` supposed to be called when raft topology changes is enabled? should we explicitly disable them?

Reply:
They should not be! We need to assert if they do.